### DISCUSSION TITLE
Custom matchers

### DISCUSSION THREAD

**Discussion by dascui** [myndpm/open-source#37] [Se habla Español]
Hola Matheo, he estado tratando de usar tu libreria y crear algunos Custom Componentes con la documentación que has dejado tanto aquí como en la pagina de MYND, sin embargo quiero implementar un contador de caracteres para los inputs y se me ocurria utilizar los matchers para poder realizar esta tarea, he estado buscando y no encuentro ninguna documentación mas que un live stream que hiciste hace un par de meses donde mencionas la posibilidad de utilizar custom matchers. 
Tienes alguna instrucciones/documentación que puedas facilitarme para realizar esta tarea, o una mejor manera de implementar este requerimiento?

**Comment by matheo** [comment]
Hola @dascui 
creo que un contador se puede implementar sencillamente imprimiendo el número de caracteres del value, algo como:
```{{ control.value?.length || 0 }}```
no habría necesidad de nada complicado a no ser que quieras hacer un procedimiento más complejo a partir de eso :)

**Reply by dascui** [reply]
Gracias por la recomendación, me estaba enfocando demasiado en la idea de usar los matches cuando en realidad no era necesario, finalmente cree modulos que fueran parecidos a los que existen y agregue el control value, gracias de nuevo y espero no haber molestado con la pregunta

**Reply by matheo** [reply]
@dascui para nada! siempre bienvenidas!
justo anoche tuve que hacer un par de matchers para esconder unos campos si una opción tenía un valor
y hacerlo requerido sólo cuando no se cumpliera esa condición:
```
function getFormConfig(featureFlag: boolean): : DynBaseConfig[] {

  // una condicion personalizada que depende de otro campo 'employmentType'
  function dynCondition(negate?: boolean): DynControlConditionFn {
    return (node: DynTreeNode) => {
      const control = node.query('employmentType');
      return control.valueChanges.pipe(
        startWith(control.value),
        map(employmentType => employmentType === MyndEmploymentType.SelfEmployed),
        map((isSelfEmployed) => {
          return featureFlag && !isSelfEmployed;
        }),
        map(result => negate ? !result : result),
      );
    };
  }

  const hideOldFields: DynControlMatch = {
    matchers: ['HIDE'],
    when: [dynCondition()],
  };
  const requiredIfNotHidden: DynControlMatch = {
    matchers: [['VALIDATE', { required: true }]],
    when: [dynCondition(true)],
  };

  return [
    createConfig('TEXT-AREA', {
      ...
      match: [hideOldField, requiredIfNotHidden],
    }),
  ];
```
con requerimientos más complejos a la hora de validar/esconder/deshabilitar los Matchers dan mucha potencia ;)
